runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -6778,23 +6779,24 @@ void checkTxCmplsCQ(struct perTxCtxInfo_t* tcip) {
 
 static
 void checkTxCmplsCntr(struct perTxCtxInfo_t* tcip) {
-  if (envProgressCntr) {
-    // Manually progress the counter
-    int rc = fi_cq_read(tcip->txCQ, NULL, 0);
-    if ((rc < 0) && (rc != -FI_EAGAIN)) {
-      INTERNAL_ERROR_V("fi_cq_read failed: %s", fi_strerror(rc));
-    }
-  }
-  uint64_t count = fi_cntr_read(tcip->txCntr);
-  if (count > tcip->numTxnsSent) {
-    INTERNAL_ERROR_V("fi_cntr_read() %" PRIu64 ", but numTxnsSent %" PRIu64,
-                     count, tcip->numTxnsSent);
-  }
-  tcip->numTxnsOut = tcip->numTxnsSent - count;
   if (tcip->numTxnsOut > 0) {
-    count = fi_cntr_readerr(tcip->txCntr);
-    if (count > 0) {
-      INTERNAL_ERROR_V("error count %" PRIu64, count);
+    if (envProgressCntr) {
+      // Manually progress the counter
+      int rc;
+      OFI_CHK_2(fi_cq_read(tcip->txCQ, NULL, 0), rc, -FI_EAGAIN);
+    }
+    OFI_CHK(fi_cntr_wait(tcip->txCntr, tcip->count + tcip->numTxnsOut, -1));
+    tcip->count = fi_cntr_read(tcip->txCntr);
```

Comment:
I may be missing something here, but why do we need this additional counter in the `tcip`? It seems like the old logic that set `numTxnsOut` to the difference between `numTxnsSent` and the current count should be enough. (In fact, we do that below.)

Reply:
You are correct, the `count` field is duplicative. I'm not sure what I was thinking. I've pushed a new version that is much simpler.